File: book-springboot/src/main/kotlin/com/gunmer/book/domain/Book.kt
package com.gunmer.book.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document("books")
class Book(
        @Id
        val id: String,
        val title: String,
        val authors: List<String>,
        val categories: List<String>,
        val isbn: String,
        val pageCount: Int,
        val status: Status,
        val thumbnailUrl: String?,
        val shortDescription: String?,
        val longDescription: String?,
        val publishedDate: Date?
)

File: book-nestjs/src/domain/model/book.ts
import { Document } from 'mongoose';

export interface Book extends Document {
  title: string;
  authors: string[];
  categories: string[];
  isbn: string;
  pageCount: number;
  status: string;
  thumbnailUrl?: string;
  shortDescription?: string;
  longDescription?: string;
  publishedDate?: Date;
}
